src/app/content/hooks/locationChange/resolveContent.ts

```diff
@@ -53,7 +53,7 @@ const resolveBook = async(
   const loader = archiveLoader.book(bookId, bookVersion);
   const state = getState();
   const bookState = select.book(state);
-  const book = bookState && bookState.id === bookId ? bookState : undefined;
+  const book = bookState && bookState.id === bookId && bookState.version === bookVersion ? bookState : undefined;
```

Comment:
I wasn't sure what 
`if a version is specified in the url, and if the canonical is defaulting to point to the current book, then it should not try to load the book again but with the version from books.config.js, it should use the loaded version.` exactly means. Is this it?

Reply:
your changes to `src/app/content/hooks/receiveContent.ts` and `src/app/content/utils/canonicalUrl.ts` are on the right track. i don't think you should have to make any changes to `src/app/content/hooks/locationChange/resolveContent.ts`